### PR TITLE
optimize pilot-discovery server

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
1. print all flags

1. remove useless `HTTPListeningAddr` return param

1. check is not found error by kubernetes standard lib instead of string match

### CODE REVIEW

File: tests/local/xds_local_test.go [hunk 1]
```diff
@@ -190,7 +190,7 @@ func initLocalPilot(IstioSrc string) (*bootstrap.Server, error) {
 
 func startLocalPilot(s *bootstrap.Server, stop chan struct{}) {
 	// Start the server
-	_, _ = s.Start(stop)
+	s.Start(stop)
```
Comment (ZackButcher) [inline tests/local/xds_local_test.go]: Hm, I'm surprised the linter doesn't complain about this; I know the `_, _` was added to make it happy.

File: pilot/pkg/bootstrap/server.go [hunk 2]
```diff
@@ -808,13 +800,13 @@ func (s *Server) initDiscoveryService(args *PilotArgs) error {
 			<-stop
 			model.JwtKeyResolver.Close()
 
-			err = s.httpServer.Close()
+			err = s.httpServer.Shutdown(context.TODO())
```
Comment (ZackButcher) [inline pilot/pkg/bootstrap/server.go]: nit: collapse into `if` below

File: tests/local/xds_local_test.go (hunk 1 above)
Comment (hzxuzhonghu) [reply tests/local/xds_local_test.go]: Not needed

File: pilot/pkg/bootstrap/server.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/bootstrap/server.go]: ok

Comment (hzxuzhonghu) [reply pilot/pkg/bootstrap/server.go]: done

**Pull request merged**